{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "20a30339",
   "metadata": {},
   "source": [
    "# Pipeline to generate Loom file for RNA Velocity analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8d2720de",
   "metadata": {},
   "source": [
    "Seurat is a very popular tool used for Single-cell Transcriptomics. The usual Seurat object or matrices used as input for Seurat generated by CellRanger do not contain the necessary information for RNA velocity analysis.\n",
    "\n",
    "[RNA velocity](https://www.nature.com/articles/s41586-018-0414-6) is estimated using the amounts of spliced and unspliced mRNAs. Common single cell feature counts, such as Cell Ranger, do not use them but it is possible to calculate them using FASTQ files.\n",
    "\n",
    "This pipeline will help you generate the files in Loom format which can be used further for RNA velocity."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ff3533fb",
   "metadata": {},
   "source": [
    "### References\n",
    "\n",
    "* https://github.com/basilkhuder/Seurat-to-RNA-Velocity#generating-loom-files\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8b955874",
   "metadata": {},
   "source": [
    "## Generate the Loom file\n",
    "\n",
    "------------------------------------------------\n",
    "###  Loom file structure\n",
    "\n",
    "<b>Loom file\n",
    "- Main layer: total counts\n",
    "- Additional layers: spliced/unspliced counts\n",
    "- Attribute/Graphs: gene names, cell IDs, embeddings"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ab6c5ab5",
   "metadata": {},
   "source": [
    "### Installation"
   ]
  },
  {
   "cell_type": "raw",
   "id": "4746a414",
   "metadata": {},
   "source": [
    "# Download dependencies first\n",
    "\n",
    "# Use your own environment to install the necessary dependencies\n",
    "python3 -m pip install numpy scipy cython numba matplotlib scikit-learn h5py click\n",
    "python3 -m pip install velocyto"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1431a8c3",
   "metadata": {},
   "source": [
    "### Test you velocyto installation through command line terminal"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "52457492",
   "metadata": {},
   "source": [
    "velocyto run --help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c4cd2b37",
   "metadata": {},
   "outputs": [],
   "source": [
    "velocyto run10x \\\n",
    "            /work3/SingleCell/yokota/20210915/E2-1/outs \\\n",
    "            /work/Database/10XGenomics/refdata-cellranger-arc-GRCh38-2020-A-2.0.0/genes/genes.gtf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5483b52f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "158b8d5b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "311e3f05",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b249e16d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ef3f83e4",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
